### PR TITLE
LWG-3850: `views::as_const` on `empty_view<T>` should return `empty_view<const T>`

### LINKED ISSUE #3418: LWG-3850 `views::as_const` on `empty_view<T>` should return `empty_view<const T>`
LWG-3850 `views::as_const` on `empty_view<T>` should return `empty_view<const T>`

### PR DESCRIPTION
**Pull request by JMazurkiewicz**
Closes #3418.

### CODE REVIEW

**Review by CaseyCarter — APPROVED** [review]

File: tests/std/tests/P2278R4_views_as_const/test.cpp [hunk 1]
```diff
@@ -303,13 +340,16 @@ constexpr bool test_one(Rng&& rng, Expected&& expected) {
     {
         const same_as<iterator_t<R>> auto i = r.begin();
         if (!is_empty) {
+            // (static analyzer doesn't realize that `i == nullptr` implies `is_empty`)
+#pragma warning(suppress : 6011) // Dereferencing NULL pointer 'i'
```
Comment (StephanTLavavej) [inline tests/std/tests/P2278R4_views_as_const/test.cpp:344]: No change requested: We avoid `suppress` in `stl/inc` (as it sometimes has issues with templates and/or modules, whereas `push/disable/pop` is reliable) but it's fine in `stl/src` and test code.

**Review by StephanTLavavej — APPROVED** [review]

**Review by strega-nil-ms — APPROVED** [review]
I was concerned about `empty_view<int&>` being wrong, but it turns out that `empty_view<reference-type>` is invalid.

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for helping the list of remaining LWG issues get closer to empty! :joy_cat: :chart_with_downwards_trend: :tada:

**Pull request merged**